js/search/controllers/search.controller.js

```diff
@@ -56,6 +84,34 @@
                 autoStart: clickedOnPlay || platform.isMobile || typeof startTime !== 'undefined'
             });
         }
+
+        /**
+         * Are there any items left to show
+         * @returns {boolean}
+         */
+        function itemsLeft() {
+            return  !!(vm.feed && (searchFeed.playlist.length > vm.feed.playlist.length));
+        }
+
+        /**
+         * @ngdoc method
+         * @name jwShowcase.search.SearchController#showMoreClickHandler
+         * @methodOf jwShowcase.search.SearchController
+         *
+         * @description
+         * Handle click event on the show more button
+         */
+        function showMoreClickHandler () {
+            var feedPlaylistLength = vm.feed.playlist.length;
+            
+            var toBeAddedMediaItems = searchFeed.playlist.slice(feedPlaylistLength, feedPlaylistLength + limit);
+
+            addItemsToFeed(toBeAddedMediaItems).then(function () {
+                vm.feed.playlist = vm.feed.playlist.concat(toBeAddedMediaItems);
+
+                $scope.$apply();
```

Comment:
Using `$q.all` makes `$scope.$apply` redundant

Reply:
Because the code in the Api service makes use of Promises instead of $q, it seems logical to use that here also.

Reply:
@ChristiaanScheermeijer I suggest I change the Promises in Api to $q

Reply:
Agreed